Flaky Test: ubuntu-latest - typedLanguageIsSet

```
AbstractFlashcardViewerTest > typedLanguageIsSet FAILED
    java.lang.AssertionError: A default model should have no preference
    Expected: null
         but: was "ja"
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at com.ichi2.anki.AbstractFlashcardViewerTest$typedLanguageIsSet$1.invokeSuspend(AbstractFlashcardViewerTest.kt:255)
        at com.ichi2.anki.AbstractFlashcardViewerTest$typedLanguageIsSet$1.invoke(AbstractFlashcardViewerTest.kt)
        at com.ichi2.anki.AbstractFlashcardViewerTest$typedLanguageIsSet$1.invoke(AbstractFlashcardViewerTest.kt)
        at com.ichi2.testutils.TestClass$runTest$1.invokeSuspend(TestClass.kt:183)
        at com.ichi2.testutils.TestClass$runTest$1.invoke(TestClass.kt)
        at com.ichi2.testutils.TestClass$runTest$1.invoke(TestClass.kt)
        at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt$runTest$2$1$1.invokeSuspend(TestBuilders.kt:314)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
        at kotlinx.coroutines.test.TestDispatcher.processEvent$kotlinx_coroutines_test(TestDispatcher.kt:28)
        at kotlinx.coroutines.test.TestCoroutineScheduler.tryRunNextTaskUnless$kotlinx_coroutines_test(TestCoroutineScheduler.kt:[103](https://github.com/ankidroid/Anki-Android/actions/runs/7345517466/job/19998787265#step:9:104))
        at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt$runTest$2$1$workRunner$1.invokeSuspend(TestBuilders.kt:320)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:[108](https://github.com/ankidroid/Anki-Android/actions/runs/7345517466/job/19998787265#step:9:109))
        at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:280)
        at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
        at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
        at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
        at kotlinx.coroutines.test.TestBuildersJvmKt.createTestResult(TestBuildersJvm.kt:13)
        at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt.runTest-8Mi8wO0(TestBuilders.kt:308)
        at kotlinx.coroutines.test.TestBuildersKt.runTest-8Mi8wO0(Unknown Source)
        at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt.runTest-8Mi8wO0(TestBuilders.kt:166)
        at kotlinx.coroutines.test.TestBuildersKt.runTest-8Mi8wO0(Unknown Source)
        at com.ichi2.testutils.TestClass.runTest(TestClass.kt:181)
        at com.ichi2.testutils.TestClass.runTest$default(TestClass.kt:175)
        at com.ichi2.anki.AbstractFlashcardViewerTest.typedLanguageIsSet(AbstractFlashcardViewerTest.kt:240)
```

--- a/AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerTest.kt
+++ b/AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerTest.kt
@@ -327,7 +327,9 @@ class AbstractFlashcardViewerTest : RobolectricTest() {
 
     private fun showNextCard(viewer: NonAbstractFlashcardViewer) {
         viewer.executeCommand(ViewerCommand.FLIP_OR_ANSWER_EASE4)
+        waitForAsyncTasksToComplete()
         viewer.executeCommand(ViewerCommand.FLIP_OR_ANSWER_EASE4)
+        waitForAsyncTasksToComplete()
     }
 
     @get:CheckResult

--- a/AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerTest.kt
+++ b/AnkiDroid/src/test/java/com/ichi2/anki/AbstractFlashcardViewerTest.kt
@@ -50,6 +50,7 @@ import java.util.*
 import java.util.stream.Stream
 import com.ichi2.anim.ActivityTransitionAnimation.Direction as Direction
 
+@Suppress("SameParameterValue")
 @RequiresApi(api = Build.VERSION_CODES.O) // getImeHintLocales, toLanguageTags, onRenderProcessGone, RenderProcessGoneDetail
 @RunWith(AndroidJUnit4::class)
 class AbstractFlashcardViewerTest : RobolectricTest() {
@@ -257,7 +258,7 @@ class AbstractFlashcardViewerTest : RobolectricTest() {
 
     @Test
     fun automaticAnswerDisabledProperty() {
-        val controller = getViewerController(true, false)
+        val controller = getViewerController(addCard = true, startedWithShortcut = false)
         val viewer = controller.get()
         assertThat("not disabled initially", viewer.automaticAnswer.isDisabled, equalTo(false))
         controller.pause()
@@ -268,7 +269,7 @@ class AbstractFlashcardViewerTest : RobolectricTest() {
 
     @Test
     fun noAutomaticAnswerAfterRenderProcessGoneAndPaused_issue9632() = runTest {
-        val controller = getViewerController(true, false)
+        val controller = getViewerController(addCard = true, startedWithShortcut = false)
         val viewer = controller.get()
         viewer.automaticAnswer = AutomaticAnswer(viewer, AutomaticAnswerSettings(AutomaticAnswerAction.BURY_CARD, true, 5, 5))
         viewer.executeCommand(ViewerCommand.SHOW_ANSWER)

